utils/s2n_init.c

```diff
@@ -89,11 +94,15 @@ static bool s2n_cleanup_atexit_impl(void)
     /* the configs need to be wiped before resetting the memory callbacks */
     s2n_wipe_static_configs();
 
-    return s2n_result_is_ok(s2n_libcrypto_cleanup()) &&
+    bool cleaned_up = \
         s2n_result_is_ok(s2n_rand_cleanup_thread()) &&
-        s2n_result_is_ok(s2n_rand_cleanup()) &&
+        s2n_result_is_ok(s2n_rand_cleanup()) &&         // not idempotent
+        s2n_result_is_ok(s2n_libcrypto_cleanup()) &&
         s2n_result_is_ok(s2n_locking_cleanup()) &&
-        (s2n_mem_cleanup() == S2N_SUCCESS);
+        (s2n_mem_cleanup() == S2N_SUCCESS);             // not idempotent
```

Comment:
I'm not sure the "not idempotent" comments are helpful / belong here. It might be worth noting which ones are / aren't idempotent on the check you added to s2n_cleanup instead.

Reply:
also `//` style comments are not allowed on all of the compilers we support.